src/nodes.coffee

```diff
@@ -4139,3 +4165,29 @@ makeDelimitedLiteral = (body, options = {}) ->
     when ps        then '\\u2029'
     when other     then (if options.double then "\\#{other}" else other)
   "#{options.delimiter}#{body}#{options.delimiter}"
+
+# Extends the location data of an AST node to include the location data from
+# another AST node.
+mergeAstLocationData = (intoNode, fromNode) ->
+  {range: intoRange} = intoNode
+  {range: fromRange} = fromNode
```

Comment:
Can you explain “into” and “from” here? I would think that this function would take input like `locationDataA`, `locationDataB` and generate a new `locationData` that includes the data from A + B. So like look at A’s `start` value and then look at B’s `start` value, and the lower (earlier) of the two becomes the merged `start` value. Ditto for `end`, and for the start/end values of the ranges.

Reply:
@GeoffreyBooth this function is mutating `intoNode`, whose existing location data should be expanded to include `fromNode`'s location data

Since we're dealing with AST nodes here (where the individual location data fields are at the same top level as other fields), returning a new "`locationData`" object seems less natural than if we were dealing with eg node class objects or tokens (both of which store `locationData` as a standalone object)

And I think it's a cleaner calling pattern for this to mutate the first node's location data fields. Otherwise it seems like you'd end up having to call it like eg
```
someExistingAstNode = Object.assign {},
  someExistingAstNode,
  mergeAstLocationData(someExistingAstNode, otherAstNode)
```
rather than just `mergeAstLocationData(someExistingAstNode, otherAstNode)`

I'm not sure what the upside would be of keeping it "immutable" and then forcing the caller to reassemble a new AST node like that, the typical case is that there's no reason to preserve the previous location data